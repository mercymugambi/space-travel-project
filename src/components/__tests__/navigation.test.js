import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../navigation';
import '@testing-library/jest-dom/extend-expect';

describe('Application interactions', () => {
  it('should render the header element', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const headerElement = screen.getByTestId('header');
    expect(headerElement).toBeInTheDocument();
  });
  it('should render the logo', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const logoElement = screen.getByAltText('Space travel hub logo');
    expect(logoElement).toBeInTheDocument();
  });
  it('should render the rocket link', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const rocketLinkElement = screen.getByRole('link', { name: /rockets/i });
    expect(rocketLinkElement).toBeInTheDocument();
  });
  it('should render the missions link', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const missionsLinkElement = screen.getByRole('link', { name: /missions/i });
    expect(missionsLinkElement).toBeInTheDocument();
  });
  it('should render the profile link', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const profileLinkElement = screen.getByRole('link', { name: /my profile/i });
    expect(profileLinkElement).toBeInTheDocument();
  });
});
