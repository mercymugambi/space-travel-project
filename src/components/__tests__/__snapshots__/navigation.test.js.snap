// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application interactions should match the snapshot 1`] = `
<header
  data-testid="header"
>
  <div
    className="logo"
  >
    <img
      alt="Space travel hub logo"
      className="logoImg"
      src="planet.png"
    />
    <h1>
      Space Traveler's Hub
    </h1>
  </div>
  <nav>
    <ul
      className="navList"
    >
      <li>
        <a
          aria-current="page"
          className="navItem rocket active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="navItem"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <div
        className="bar"
      />
      <li>
        <a
          className="navItem"
          href="/Profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
